### PR TITLE
Settings menu options - Presenter/Moderator vs Viewer

### PR DESCRIPTION
**Pull request by KDSBrowne**
The settings menu now renders options based on weather the user is a Presenter/Moderator or Viewer.

Audio, Video and Application setting options can be seen by all users.

Participants setting option is only visible to Presenter/Moderator.

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/settings/component.jsx [hunk 1]
```diff
@@ -14,29 +14,56 @@ export default class Settings extends React.Component {
...
+        props: { title: 'Participants', prependIconName: 'icon-', icon: 'bbb-user', }, });
+    }
+
+    return (
+      <div className={styles.full} role='presentation'>
+        <div className={styles.settingsMenuLeft}>
+          <ul className={styles.settingsSubmenu} role='menu'>
+            {this.submenus.map((value, index) => (
+              <li key={index} ref={'submenu' + index} role='menuitem' tabIndex={value.tabIndex}
+                onClick={this.clickSubmenu.bind(this, index)}
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/settings/component.jsx]: Change `clickSubmenu` to `handleClick` or `handleClickSubmenu`

File: bigbluebutton-html5/imports/ui/components/settings/component.jsx [hunk 2]
```diff
@@ -14,29 +14,56 @@ export default class Settings extends React.Component {
...
+          </ul>
+        </div>
+        <div className={styles.settingsMenuRight} role='presentation'>
+          {this.createMenu()}
+        </div>
+      </div>
+    );
   }
 
   createMenu() {
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/settings/component.jsx]: `renderMenu`

File: bigbluebutton-html5/imports/ui/components/settings/component.jsx [hunk 3]
```diff
@@ -14,29 +14,56 @@ export default class Settings extends React.Component {
...
-     * initialized to 0
-     */
-    this.setState({ activeSubmenu: 0 });
-    /* focusSubmenu represents the submenu in the submenus array which currently has focus,
-     * initialized to 0
-     */
-    this.setState({ focusSubmenu: 0 });
+  renderSettingOptions() {
+    const { isPresenter, role } = this.props;
+    this.submenus = [];
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/settings/component.jsx]: You should move those `push's` to the `constructor` function. The render function should only contain render related logic.

And you can push multiple objects in a single push: `this.submenus.push({...}, {...}, {...}, ...)`

File: bigbluebutton-html5/imports/ui/components/settings/component.jsx [hunk 4]
```diff
@@ -14,29 +14,76 @@ export default class Settings extends React.Component {
...
+          {this.renderMenu()}
+        </div>
+      </div>
+    );
+  }
+
+componentWillReceiveProps(nextProps) {
+  this.submenus = [];
+
+  this.submenus.push(
```
Comment (Gcampes) [inline bigbluebutton-html5/imports/ui/components/settings/component.jsx]: Why do we need this in 2 places? wouldn't it be better if this was inside `render`? this way it wouldn't need to be in 2 different places

Comment (KDSBrowne) [inline bigbluebutton-html5/imports/ui/components/settings/component.jsx]: That's how I had it a first. It was suggested to take it out of the `render `.

Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/settings/component.jsx]: Ops, my bad :confounded:. Didn't notice that the menus are using props to be populated and need to change when the component update. Sorry,

**Pull request merged**